File: codeforces/382_C.py
n = int(input())
if n == 1:
    print(-1)
    exit
else:
    arr = [*map(int, input().split())]
    arr.sort();
    ans = set();
    freq = {};
    for i in range(1, n):
        if arr[i] - arr[i-1] not in freq:
            freq[arr[i] - arr[i-1]] = 1
        else:
            freq[arr[i] - arr[i-1]] += 1


    if len(freq) == 1:
        ans.add(arr[0] - arr[1] + arr[0]);
        if n == 2 and (arr[1] - arr[0]) % 2 == 0:
            ans.add(arr[0] + (arr[1] - arr[0]) / 2);
        
        ans.add(arr[n-1] + (arr[1] - arr[0]));
    if len(freq) == 2:
        freq = list(freq.items())
        freq.sort()
        if freq[1][1] == 1:
            for i in range(1, n):
                if arr[i] - arr[i-1] == freq[1][0] and (arr[i] - arr[i-1]) % 2 == 0 and (arr[i] - arr[i-1]) / 2 == freq[0][0]:
                    ans.add((arr[i] + arr[i-1]) / 2);
                    break;

    ans = [*map(int, ans)]
    ans = [*map(str, ans)]
    print(len(ans));
    print(" ".join(ans));

File: codeforces/382_C.cpp
#include <bits/stdc++.h>

using namespace std;
using ll = long long;

int main(){
    int n; cin >> n;
    if(n == 1){
        cout << -1;
        return 0;
    }
    ll arr[n];
    for(ll &a: arr) cin >> a;
    sort(arr, arr + n);
    set<int> ans;
    map<int,ll> freq;
    for(int i = 1; i < n; i++){
        freq[arr[i] - arr[i-1]]++;
    }
    
    if(freq.size() == 1){
        ans.insert(arr[0] - arr[1] + arr[0]);
        if(n == 2 && (arr[1] - arr[0]) % 2 == 0){
            ans.insert((ll)arr[0] + (arr[1] - arr[0]) / 2);
        }
        ans.insert(arr[n-1] + (arr[1] - arr[0]));
    }else if(freq.size() == 2){
        auto it = freq.begin();
        it++;
        if(it->second == 1){
            for(int i = 1; i < n; i++){
                if(arr[i] - arr[i-1] == it->first && (arr[i] - arr[i-1]) % 2 == 0 && (arr[i] - arr[i-1]) / 2 == (freq.begin())->first){
                    ans.insert((arr[i] + arr[i-1]) / 2);
                    break;
                }
            }
        }
    }
    cout << ans.size() << '\n';
    for(int i: ans) cout << i << ' ';
    return 0;
}